### PR TITLE
Implement conversion of legacy composer groups

### PR DESCRIPTION
**Pull request by manisandro**
Implement conversion of legacy composer groups

### CODE REVIEW

**Comment by nyalldawson** [comment]
Code looks good! Can you add a test please?

**Comment by manisandro** [comment]
@nyalldawson How does one generate the control images for the tests in `/tests/testdata/control_images/compositionconverter`?

**Comment by nyalldawson** [comment]
> How does one generate the control images for the tests in /tests/testdata/control_images/compositionconverter?

I don't think a rendering test is required here -- that won't reveal if groups are recreated or not. I'd suggest just directly checking the resulting items present in the created layout.

**Comment by manisandro** [comment]
Ok, done!

File: tests/src/core/testqgscompositionconverter.cpp [hunk 1]
```diff
@@ -543,6 +549,29 @@ void TestQgsCompositionConverter::importComposerTemplateScaleBar()
...
+  QCOMPARE( layout->pageCollection()->pageCount(), 1 );
+
+  QList<QgsLayoutItemGroup *> items;
+  layout->layoutItems<QgsLayoutItemGroup>( items );
+  QCOMPARE( items.size(), 1 );
+
+  QgsLayoutItemGroup *item = items.at( 0 );
+  QVERIFY( item->isVisible() );
+
+  qDeleteAll( items );
```
Comment (nyalldawson) [inline tests/src/core/testqgscompositionconverter.cpp:571]:
```suggestion
```

layoutItems doesn't transfer ownership

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Thanks @manisandro !

**Pull request merged**